### PR TITLE
186 session subscriberid

### LINKED ISSUE #185: Session Subscriber
Store SubscriberID in the current user Session and use that to determine which Subscriber the User is acting on behalf of.

For this ticket, just grab the first Subscriber that is affiliated with the Current User. Eventually, we'll support multiple Subscribers per user and we'll present a mechanism to update the Session Subscriber the User is acting on behalf of in the UI. At this time, we'll just support one and thus we'll grab he first.

### PR DESCRIPTION
**Pull request by JPrevost**
Please see individual commit messages for some details on this PR.

Closes #185

### CODE REVIEW

**Comment by JPrevost** [comment]
FWIW: the Travis build crashed since I reenabled test in Fork which I needed for local development. There is an open Issue with the Travis team on this and they suggested trying the old architecture by setting `sudo: required` in travis.yml. I have opened a new branch with this setting. If it works, I'll PR that as well. If it still fails, we can investigate whether or not there is a way to fork locally in Test but not on Travis... or just stop caring about Travis for now. Or I can make that change locally when I need it I guess.

**Comment by JPrevost** [comment]
2fcf82e merged into this PR a Travis setting to use the old infrastructure (non container based) which does seem to be a workaround for now at least.

File: app/controllers/Application.scala [hunk 1]
```diff
@@ -365,6 +355,7 @@ object Application extends Controller with Security {
   }
 
   def scheme(id: Int) = isAnalyst { identity => implicit request =>
+    // todo: update to currentSubscriberId
```
Comment (JPrevost) [inline app/controllers/Application.scala]: Oops, missed one. I'll add a commit to this PR.

Comment (richardrodgers) [inline app/controllers/Application.scala]: Umm, I might want to rethink this. Right now the scheme page is where subscribers add interests, but generally its where users can learn about the scheme.
I don’t see an analyst restriction (although there *are* analyst-specific functions that might link from here)

RIchard

Comment (JPrevost) [inline app/controllers/Application.scala]: I noticed that as well just now. I think until you added Interests 2.0 to that page it seemed like just a page with the Scheme name and link to adminy things so the `isAnalyst` restriction made sense to me. With that being the only entry point to add a new Interest, we do indeed need a ticket to update the required role.

Comment (JPrevost) [inline app/controllers/Application.scala]: https://github.com/MITLibraries/scoap3hub/issues/190

**Comment by richardrodgers** [comment]
:+1: for merging. I would like in time to begin a refactor over test code so that the browser login actions are abstracted somewhat, so we can change the mechanism without touching all the test classes, but that and other abstractions should be tackled in a different thread/set of issues

**Comment by JPrevost** [comment]
We can probably clean the tests up with Traits... but yeah, another day.

**Pull request merged**